streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -261,7 +311,14 @@ public void print(final Printed<K, V> printed) {
         allSourceNodes.addAll(sourceNodes);
         allSourceNodes.addAll(streamImpl.sourceNodes);
 
-        builder.internalTopologyBuilder.addProcessor(name, new KStreamPassThrough<>(), parentNames);
+        ProcessorParameters processorParameters = new ProcessorParameters<>(new KStreamPassThrough<>(), name);
+
+        StatelessProcessorNode<K, V> mergeNode = new StatelessProcessorNode<>(name,
```

Comment:
And regarding to my meta comment on the first PR: after reviewing this one it still seems that we have the parent graph nodes all already created before creating the children nodes, so why not just passing in the parent node references directly in constructor, and hence not needing to get it from the parent node any more???

Reply:
refactored to include graph nodes in constructor calls.